lib/ofex.ex

```diff
@@ -47,14 +47,21 @@ defmodule Ofex do
   def parse(data) do
     case validate_ofx_data(data) do
       {:ok, parsed_ofx} ->
-        parsed_ofx
+        result = parsed_ofx
         |> xpath(~x"//OFX/*[contains(name(),'MSGSRS')]"l)
         |> Enum.map(&parse_message_set(xpath(&1, ~x"name()"s), &1))
-        |> Map.new
+        |> List.flatten
+        |> Enum.reduce(%{signon: %{}, accounts: []}, &reduce_items_list/2)
+
+        {:ok, result}
       {:error, message} -> {:error, %InvalidData{message: message, data: data}}
     end
   end
 
+  defp reduce_items_list(%{signon: signon}, %{accounts: accounts}), do: %{signon: signon, accounts: accounts}
+  defp reduce_items_list(%{account: account}, %{accounts: accounts}=acc), do: Map.put(acc, :accounts, [account | accounts])
+  defp reduce_items_list(_, acc), do: acc
```

Comment:
Not totally in love with the function name, but not sure I have a better idea. Maybe something like `accumulate_parsed_items`?